Modules/_testcapimodule.c

```diff
@@ -3888,6 +3888,23 @@ with_tp_del(PyObject *self, PyObject *args)
     return obj;
 }
 
+static PyObject *
+without_gc(PyObject *Py_UNUSED(self), PyObject *obj)
+{
+    PyTypeObject *tp = (PyTypeObject*)obj;
+    if (!PyType_Check(obj) || !PyType_HasFeature(tp, Py_TPFLAGS_HEAPTYPE)) {
+        return PyErr_Format(PyExc_TypeError, "heap type expected, got %R", obj);
+    }
+    if (PyType_IS_GC(tp)) {
+        // Don't try this at home, kids:
+        tp->tp_flags -= Py_TPFLAGS_HAVE_GC;
+        tp->tp_free = PyObject_Del;
```

Comment:
Not that matters much, but just to not get confused if this ever pops when debugging something in the gc:

```suggestion
        tp->tp_flags -= Py_TPFLAGS_HAVE_GC;
        tp->tp_traverse = NULL;
        tp->tp_clear = NULL;
        tp->tp_free = PyObject_Del;
```